src/features/Slices/postSlise.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import * as postsApi from '../../api/posts';
+import { Post } from '../../types/Post';
+
+export interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  async (userId: number) => {
+    const posts = await postsApi.getUserPosts(userId);
+
+    return posts;
+  },
+);
```

Comment:
You've implemented the `fetchPosts` function with `createAsyncThunk`. This is a good use of Redux Toolkit's async functions. However, it would be a good idea to handle potential errors that could occur during the API call. You can do this by wrapping the `await postsApi.getUserPosts(userId)` call in a try-catch block. This way, if the API call fails for any reason, you can handle the error gracefully and avoid a potential application crash.